demo/midgard-sdk/src/tx-builder/user-events/deposit/index.ts

```diff
@@ -1 +1,96 @@
+import {
+  Data,
+  Integer,
+  LucidEvolution,
+  RedeemerBuilder,
+  Script,
+  toUnit,
+  TransactionError,
+  TxBuilder,
+} from "@lucid-evolution/lucid";
+import { Datum, DepositInfo, MintRedeemer } from "./types.js";
+import { POSIXTime } from "@/tx-builder/common.js";
+import { bufferToHex, hashHexWithBlake2b256 } from "@/utils/common.js";
+import { Effect } from "effect";
+
+export type DepositParams = {
+  depositScriptAddress: string;
+  mintingPolicy: Script;
+  policyId: string;
+  depositInfo: DepositInfo;
+  inclusionTime: POSIXTime;
+};
+
+/**
+ * Deposit
+ *
+ * @param lucid - The LucidEvolution
+ * @param params - The parameters
+ * @returns {TxBuilder} A TxBuilder instance that can be used to build the transaction.
+ */
+
+export const depositTxBuilder = (
+  lucid: LucidEvolution,
+  params: DepositParams,
+): Effect.Effect<TxBuilder, TransactionError | unknown> =>
+  Effect.gen(function* () {
+    const utxos = yield* Effect.promise(() => lucid.wallet().getUtxos());
+    if (utxos.length === 0) {
+      throw new Error("No UTxOs found in wallet");
```

Comment:
In `effect` there is no need for `throw`s.
```suggestion
      yield* new LucidError({message: "Failed to build the deposit transaction", cause: "No UTxOs found in wallet"});
```